src/geometry.rs

```diff
@@ -272,6 +255,7 @@ impl<U, T: Add<U>> Add<Size<U>> for Size<T> {
     }
 }
 
+#[allow(dead_code)]
```

Comment:
It's because these are helper methods for thing like `with_main` and `with_cross`, and we want to provide the same methods for each axis. But in some cases we only actually have a use case for one of them.